pkg/antctl/raw/multicluster/common/common.go

```diff
@@ -0,0 +1,425 @@
+			return err
+		}
+		fmt.Fprintf(cmd.OutOrStdout(), "ServiceAccount \"%s\" deleted in Namespace %s\n", serviceAccount.Name, namespace)
+	}
+
+	return nil
+}
+
+func CreateMemberToken(cmd *cobra.Command, k8sClient client.Client, name string, namespace string, file *os.File, createdRes *[]map[string]interface{}) error {
+	var createErr error
+	serviceAccount := newServiceAccount(name, namespace)
+	unstructuredSA, _ := runtime.DefaultUnstructuredConverter.ToUnstructured(serviceAccount)
+
+	createErr = k8sClient.Create(context.TODO(), serviceAccount)
+	if createErr != nil {
+		if !apierrors.IsAlreadyExists(createErr) {
+			fmt.Fprintf(cmd.OutOrStdout(), "Failed to create ServiceAccount \"%s\", error: %s\n", name, createErr.Error())
+			return createErr
+		}
+		fmt.Fprintf(cmd.OutOrStderr(), "ServiceAccount \"%s\" already exists\n", name)
+		createErr = nil
+	} else {
+		fmt.Fprintf(cmd.OutOrStdout(), "ServiceAccount \"%s\" created\n", serviceAccount.Name)
+		*createdRes = append(*createdRes, unstructuredSA)
+	}
+
+	roleBinding := newRoleBinding(name, name, namespace)
+	unstructuredRoleBinding, _ := runtime.DefaultUnstructuredConverter.ToUnstructured(roleBinding)
+
+	createErr = k8sClient.Create(context.TODO(), roleBinding)
+	if createErr != nil {
+		if !apierrors.IsAlreadyExists(createErr) {
+			fmt.Fprintf(cmd.OutOrStdout(), "Failed to create RoleBinding \"%s\", error: %s\n", name, createErr.Error())
+			return createErr
+		}
+		fmt.Fprintf(cmd.OutOrStderr(), "RoleBinding \"%s\" already exists\n", name)
+		createErr = nil
+	} else {
+		fmt.Fprintf(cmd.OutOrStdout(), "RoleBinding \"%s\" created\n", roleBinding.Name)
+		*createdRes = append(*createdRes, unstructuredRoleBinding)
+	}
+
+	secret := newSecret(name, name, namespace)
+	createErr = k8sClient.Create(context.TODO(), secret)
+	if createErr != nil {
+		if !apierrors.IsAlreadyExists(createErr) {
+			fmt.Fprintf(cmd.OutOrStdout(), "Failed to create Secret \"%s\", start rollback\n", name)
+			return createErr
+		}
+		fmt.Fprintf(cmd.OutOrStderr(), "Secret \"%s\" already exists\n", name)
+	}
+	// It will take one or two seconds to wait for the Data.token to be created.
+	if err := waitForSecretReady(k8sClient, name, namespace); err != nil {
+		return err
+	}
+	fmt.Fprintf(cmd.OutOrStdout(), "Secret \"%s\" created\n", secret.Name)
+
+	if file == nil {
+		return nil
+	}
+
+	if err := k8sClient.Get(context.TODO(), types.NamespacedName{Name: name, Namespace: namespace}, secret); err != nil {
+		return err
+	}
+	s := &corev1.Secret{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "v1",
+			Kind:       "Secret",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+		Data: secret.Data,
+		Type: corev1.SecretTypeOpaque,
+	}
+
+	b, err := yaml.Marshal(s)
+	if err != nil {
+		return err
+	}
+	if _, err := file.Write([]byte("\n# Create a token Secret with the embedded manifest.\n")); err != nil {
```

Comment:
I feel this line is not necessary as it looks strange in the standalone token Secret YAML file (not the join-config YAML). But not a big deal.

Reply:
If we want to keep it, maybe rephrase the comment to be more generic, like

"# Manifest to create a Secret for a member cluster token."

Reply:
Changed the comment to " Manifest to create a Secret for a member cluster token."